src/main/kotlin/no/skatteetaten/aurora/boober/feature/ToxiproxySidecarSpec.kt

```diff
@@ -26,30 +30,35 @@ data class ToxiproxyConfig(
     val enabled: Boolean = true
 )
 
-internal val AuroraDeploymentSpec.toxiproxyVersion: String?
-    get() = featureEnabled("toxiproxy") { this["toxiproxy/version"] }
-
-// Find the variable name in a field name
-internal fun findVarNameInFieldName(urlSource: ToxiproxyUrlSource, fieldName: String): String =
-    Regex("(?<=^toxiproxy\\/${urlSource.propName}\\/)([^\\/]+(?=\\/enabled\$|\\/proxyname\$|\\/initialEnabledState\$|\$))")
-        .find(fieldName)!!
-        .value
+private data class ToxiproxyEndpoint(
+    val proxyName: String,
+    val varName: String,
+    val enabled: Boolean
+)
 
-// Find the proxy name in a server and port field name
-internal fun findProxyNameInServerAndPortFieldName(fieldName: String) =
-    Regex("(?<=^toxiproxy\\/serverAndPortFromConfig\\/)([^\\/]+(?=\\/serverVariable\$|\\/portVariable\$|\\/initialEnabledState\$|\$))")
-        .find(fieldName)!!
-        .value
+private data class ToxiproxyServerAndPortVars(
+    val proxyname: String,
+    val serverVar: String,
+    val portVar: String,
+    val enabled: Boolean
+)
 
-// Generate a default proxy name based on the variable name
-// To be used when there is no proxy name specified by the user
-internal fun generateProxyNameFromVarName(varName: String, urlSource: ToxiproxyUrlSource): String =
-    "${urlSource.defaultProxyNamePrefix}_$varName"
+internal val AuroraDeploymentSpec.toxiproxyVersion: String?
+    get() = featureEnabled(FEATURE_NAME) { this[ToxiproxyField.version] }
+
+internal fun AuroraDeploymentSpec.validateToxiproxy(): List<AuroraDeploymentSpecValidationException> {
+    return listOf(
+        missingOrInvalidEndpointVariableErrors(),
+        missingServerAndPortVariableErrors(),
+        missingDbErrors(),
+        proxynameDuplicateErrors()
+    ).flatten()
+}
 
 // Validate that for every endpoint in toxiproxy/endpointsFromConfig, there is a corresponding environment variable
 internal fun AuroraDeploymentSpec.missingOrInvalidEndpointVariableErrors() =
-    groupToxiproxyEndpointFields().keys.mapNotNull { varName ->
-        val envVar = this.getSubKeys("config")["config/$varName"]?.value<String>()
+    getToxiproxyEndpointEnvVars().mapNotNull { varName ->
+        val envVar = this.getOrNull<String>("config/$varName")
```

Comment:
Ikke direkte relatert til denne PR-en, men kanskje vi burde ha noe dokumentasjon på generelle `AuroraDeploymentSpec`-funksjoner som f.eks. `getOrNull` og `getSubKeys` for å gjøre livet enklere for fremtidige Boober-utviklere? Eller finnes dette allerede?

Reply:
Jeg tror ikke det finnes. Jeg tar det for gitt selv siden jeg er vandt til å bruke de. 

Det er nok lurt hvis du tenker det ikke er tydelig fra andre features, da er det jo tegn på at det mangler. 
Hvor hadde vært naturlig for du å se eller lete etter det? :)

Reply:
Jeg ser for meg en Confluence-side, med lenke fra readme, sånn at den er lett å finne. Ev. bare ha det direkte i readme, kanskje det er lettere å følge det opp da. Kan gjerne også ha med grunnleggende informasjon om de forskjellige stegene i Feature-algoritmen, altså `createContext`, `handlers` osv. Ser det er kommentarer på disse i Feature-interfacet, men jeg tror det hadde vært fordelaktig med en mer grundig beskrivelse av rekkefølgen ting skjer i og noen eksempler. 🙂

Reply:
Jeg er litt splittet egentlig om det burde være readme eller confluence. Readme idag er rettet mest mot andre utenfor huset virker det for meg. Kanskje vi burde ha en wiki side om utvikling av boober slik du sier. Problemet er vel å holde den ved like, men kanskje slike generelle ting hører hjemme der. Jeg kan lage en oppgave på det